crates/rpc/src/v05/method/trace_block_transactions.rs

```diff
@@ -81,26 +80,44 @@ pub async fn trace_block_transactions(
     context: RpcContext,
     input: TraceBlockTransactionsInput,
 ) -> Result<TraceBlockTransactionsOutput, TraceBlockTransactionsError> {
-    let block_id = match input.block_id {
-        BlockId::Pending => return Err(TraceBlockTransactionsError::BlockNotFound),
-        other => other.try_into().expect("Only pending cast should fail"),
-    };
-
-    let (transactions, gas_price, parent_block_hash): (Vec<_>, Option<U256>, BlockHash) = {
-        let span = tracing::Span::current();
-
-        let storage = context.storage.clone();
-        tokio::task::spawn_blocking(move || {
-            let _g = span.enter();
-            fetch_block_transactions(storage, block_id)
-        })
-        .await
-        .context("trace_block_transactions: fetch block & transactions")??
+    let (transactions, gas_price, parent_block_hash) = match input.block_id {
+        BlockId::Pending => {
+            if let Some(pending) = &context.pending_data {
+                let block = pending
+                    .block()
+                    .await
+                    .ok_or(TraceBlockTransactionsError::BlockNotFound)?;
```

Comment:
Is there an argument to be made that we should instead simply return an empty trace? Although that could be confusing while still syncing etc I guess.

On second thought; this is probably best. We can revisit if we get complaints.

Reply:
We can look into that -- but this time we should clearly document behavior for all methods ;)